### DISCUSSION TITLE
Are all .dll files absolutely necessary to run ACE3?

### DISCUSSION THREAD

**Discussion by Robin-728** [acemod/ACE3#10353] [Q&A]
Hello, I'm currently hosting an Antistasi server for a few friends, however, one of them is playing on GeforceNow. 
I did some reading and I've heard that it's due to the .dll files, so I'm looking at a way to remove the DLL files (and all files/modules that are dependent on them!)
Would this be possible? If not, is there any version of ACE3 where this could be possible, no matter how old/outdated/deprecated?

**Comment by johnb432** [comment]
It should (in theory) be possible.

`advanced_ballistics`, `arsenal`, `artillerytables`, `atragmx`, `common`, `fcs`, `interact_menu`, `optionsmenu`, `rangecard`, `scopes` and `xm157` require the .dll.

It means you'd need to at least disable all of the artillery stuff that ACE does, as well as the advanced ballistics. If you force the interaction menus to be lists, then I think you can avoid the extension usage in `interact_menu`. `common` checks the extension version, so you can remove that fairly easily. `optionsmenu` only requires it for debug information.

Take a look at https://ace3.acemod.org/wiki/development/dependencies, to see what modules can be removed. If you want to keep specific modules that use the .dll, I suggest searching for `callExtension` within all of ACE' code, then determining yourself how much effort you want to put in to remove the extension from the code.
